submission/races.go

```diff
@@ -113,51 +102,41 @@ func (sub *safeSubmissionState) setResult(logURL string, sct *ct.SignedCertifica
 		sub.results[logURL] = &submissionResult{sct: sct, err: err}
 		return
 	}
+	// group name associated with logURL outside of BaseName.
+	// (this assumes the logURL is associated with only one group ignoring BaseName)
 	// If at least one group needs that SCT, result is set. Otherwise dumped.
 	for groupName := range sub.logToGroups[logURL] {
 		// Ignore the base group (All-logs) here to check separately.
 		if groupName == ctpolicy.BaseName {
 			continue
 		}
-		if sub.groupNeeds[groupName] > 0 {
+		// Set the result if the group does not have a submission.
+		if sub.groupsSubmitted[groupName] == 0 {
 			sub.results[logURL] = &submissionResult{sct: sct, err: err}
+			sub.groupsSubmitted[groupName]++
```

Comment:
This will only be executed for the first submission to this group, and therefore will never be larger than 1. Do you mean to use groupsSubmitted to implement a set?

Reply:
Ah yes this should be a set. I have changed groupsSubmitted to a map[string]bool.